test/Oscoin/Test/Storage/Block/Orphanage.hs

```diff
@@ -67,25 +61,24 @@ propSelectBestCandidateSingleChoice Dict = do
         showIt       = showOrphans . (initialChain,)
         forkParams = ForkParams 0 10 10  -- 10 fork of max 10 blocks.
     forAllShow (genOrphanChainsFrom forkParams initialChain) showIt $ \chains ->
-        monadic runIdentity $ do
-            ps <- forM chains $ \(orphanChain, missingLink) -> do
-                -- First of all, we insert all the orphans up to the missing
-                -- link, and we assess that the best chain must be 'Nothing',
-                -- as we don't have any candidate linking to the main chain.00
-                let orphanage' = insertOrphans (toNewestFirst $ blocks orphanChain) orphanage
-                let p1 = selectBestChain [blockHash defaultGenesis] orphanage' === Nothing
-                -- Now we insert the missing link and we check the chain is
-                -- what we expect.
-                let orphanage'' = insertOrphans [missingLink] orphanage'
-                let bestChain = toList
-                              . toNewestFirst
-                              . reverse
-                              . toBlocksOldestFirst orphanage''
-                              . snd <$> selectBestChain [blockHash defaultGenesis] orphanage''
-                let p2 = bestChain === Just ((toNewestFirst . blocks) orphanChain <> [missingLink])
-                pure [p1, p2]
-            pure $ classifyChainsByScore chains $
-                       foldl' (.&&.) (property True) (mconcat ps)
+            classifyChainsByScore chains
+                $ conjoin
```

Comment:
I think the reason why I had added the monadic code in the first place was that it makes slightly easier to pinpoint errors, as `conjoin` will simply compose everything together but that won't necessary help to isolate the failing property (in term of stacktracking). Would you do a simple test for me? Try introducing a (trivial) failure in one of the properties, and compare the stacktraces you get from one or the other piece of code -- if they are equivalent, then I'm fine with that 👍

Reply:
I checked: Using `monadic runIdentity` and `.&&.` over `conjoin` does not give us anything. The error reporting is improved by `===` from `Test.Quickcheck.Extended`.

Reply:
Perfect 👍